### ISSUE TITLE
Slow first connection after PM2 restart

### ISSUE THREAD

**Issue by Abashevav** [MonetDB/monetdb-nodejs#24]
In addition to the problem described in issues [#23](https://github.com/MonetDB/monetdb-nodejs/issues/23), we constantly get another problem.

We are using `_pm2_` manager for production Node.js application. `pm2` has default settings.
_MonetDB v_ 11.45.7-20220923
The _ecosystem.config.js_ file has the following settings:

```
module.exports = {
  apps: [
    {
      name: 'csr-next',
      script: 'npm',
      args: 'start',
      out_file: '/media/data/var/log/cht.log',
      error_file: '/media/data/var/log/debug-cht.log',
      env: {
        PORT: 8024
      }
    }
  ]
};
```

We make regular requests to MonetDB. Request processing and response are fast (200-300 ms). We see this in the logs and in Postman.

BUT after we run `pm2` restart (_`pm2 restart csr-next`_ comand) the next request responds very slowly. The response to the request may return in 30-50 seconds, or may even return a Timeout error.
The log file after reboot shows:
```
> start
> next start -p PORT

ready - started server on 0.0.0.0:8024, url: http://localhost:8024
info  - Loaded env from /media/data/home/csr-next/.env.production.local
WARNING Reconnect attempt 1/10 in 2 sec..
WARNING Reconnect attempt 1/10 in 2 sec..
WARNING Reconnection succeeded.
WARNING Reconnection succeeded.
```

In this case, the number of WARNING (Reconnect + Reconnection) can be different from 1 pair to 7.
It turns out that after restarting `pm2` this library have problem with get the first connection? 
But even after the `_WARNING Reconnection succeeded._` message, the request **is still not answered**.
If we send a request again, then everything works as usual, until the next reboot of `pm2`.

Any ideas why this is happening?
How can we avoid "losing" the first requests after `pm2` restart?

**Comment by sstalin** [comment]
When application reconnects after reboot the previous connection socket gets destroyed, hence you will lose the answer from the previous request. What is not worth investigating here why the first connection attempt is slow for you after reboot.

**Comment by Abashevav** [comment]
It turns out that it is not possible to set up a "Graceful" restart for such a project?
(Since the previous request will be "lost", in case of restarting PM2)

How do we investigate why our first connection attempt is slow right after a reboot?

**Comment by sstalin** [comment]
Bug fixed in MonetDB default branch, see  issue#23 comments.
